Source/WebGPU/WGSL/AST/Statements/VariableStatement.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2022 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include "Decl.h"
+#include "Statement.h"
+#include <wtf/UniqueRef.h>
+
+namespace WGSL::AST {
+
+class VariableStatement final : public Statement {
+    WTF_MAKE_FAST_ALLOCATED;
+public:
+    VariableStatement(SourceSpan span, VariableDecl&& decl)
+        : Statement(span)
+        , m_decl(makeUniqueRef<VariableDecl>(WTFMove(decl)))
+    {
+    }
+
+    Kind kind() const override { return Kind::Variable; }
+    Decl& decl() { return m_decl.get(); }
+
+private:
+    UniqueRef<Decl> m_decl;
```

Comment:
What's the point of this class? Why not just put an enum inside `Decl`?

Reply:
I thought about this too, but it's an adapter from `Statement` to `Decl`. A `CompoundStatement` holds `Statements` and one of those statement kinds is `VariableStatement` that is going hold a `Decl` for `var`, `let`, `const` etc.

My current position is to implement that AST following the WGSL grammar in the spec. We can "optimise" the AST later. What do you think?

Reply:
Sounds good to me.